### PR TITLE
This adds a common geometry utility module.

### PR DESCRIPTION
**Pull request by bobbyesh**
## Overview

This module implements the geometry structures and methods layed out in the dot aych file.

Two points of deviation from the original document include:

1. Points are defined with float (x, y) values instead of i32, as some of the algorithms will need greater precision than integers offer.  Although Cairo used the cairo_fixed_t type, which is an integer, you can see the discussion [here](http://pixman.freedesktop.narkive.com/YV6HUW6g/cairo-floating-point-api-in-pixman) on their consideration for converting to all `double` values instead of the fixed type.  The main reason they don't use `double` is because old ARM processors don't have a floating point processing unit, and so operations on floats are slower on these machines than integers.  So instead these `cairo_fixed_t` types are tossed to Pixman as fixed point precision numbers, where the first section of bits internal to that int refer to the integer value, and the remaining bits refer to the fractional part (for more precision).  Since we aren't using Pixman and probably won't see Cairus compiled onto ARM, and would probably need to hand tune for ARM anyway, I thought that doing this bitwise processing for points would be overkill for our immediate purposes.
2. I did not implement a `Trapezoids` struct, as I didn't know what operations it would need yet that a `Vec<Trapezoid>` wouldn't accommodate already.  Of course, I'm sure they exist.  I assumed it would be more efficient for those operations to be added as they are needed. 

## Future

I think that we should add to common_geometry.rs as we need more functionality of out of these structs than I provide here.

## Reference

The contents of the dot ayche file:

Point
- 2 signed i32 types

Line
- Midpoint?
- slope?
 
- Knot
- 4 points
 
Vector
- Add
- angle
 
Trapezoid 
- 4 points

Trapezoids

### CODE REVIEW

**Comment by bobbyesh** [comment]
Removed Trapezoid because I realized it will only be used inside the trapezoid rasterizer, so isn't actually a common struct.  See the trapezoid-rasterizer branch for that struct.

**Comment by bobbyesh** [comment]
Slope needs to be an option, for division by zero.

File: src/common_geometry.rs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    fn eq(&self, other: &Point) -> bool {
+        self.x == other.x && self.y == other.y
+    }
+}
+
+/// ## Line
+///
+/// Defines a line by two points.
+#[derive(Debug, Copy, Clone)]
+pub struct Line {
```
Comment (kneitinger) [inline src/common_geometry.rs]: Change to LineSegment to avoid ambiguity

File: src/common_geometry.rs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+    // Returns a line.  Constructed from two points.
+    pub fn from_points(point1: Point, point2: Point) -> Line {
+        Line {
+            point1: point1,
+            point2: point2,
+        }
+    }
+
+    pub fn get_slope(&self) -> f32 {
```
Comment (kneitinger) [inline src/common_geometry.rs]: Avoid divide by zero

**Comment by bobbyesh** [comment]
After writing the trapezoid thing, I realized that although I needed a lot from here, I needed to make a lot of changes and add a lot of methods.  So... this module is probably overkill.  How about we just implement everything we need to implement as we are writing our sections, and if there happen to be any common geometry structs in the end, then we go had and collect them into a shared module?

**Pull request merged**